src/components/Taxonomy/Taxonomy.tsx

```diff
@@ -307,7 +346,18 @@ const TaxonomyDropdown = ({ show, flatten, items, dropdownRef }: TaxonomyDropdow
         onInput={onInput}
         ref={inputRef}
       />
-      {list.map(item => <Item key={item.label} item={item} isFiltering={search !== ""} />)}
+      {show && (
+        <TreeStructure 
+          items={list} 
+          rowComponent={Item} 
+          flatten={search !== ""} 
```

Comment:
search results should be in a tree-like view, so we don't need this parameter anymore